qutebrowser/browser/greasemonkey.py

```diff
@@ -196,11 +195,16 @@ def scripts_for(self, url):
         """
         if url.scheme() not in self.greaseable_schemes:
             return MatchingScripts(url, [], [], [])
-        match = functools.partial(fnmatch.fnmatch,
-                                  url.toString(QUrl.FullyEncoded))
+
+        def _match(pattern):
+            display_url = url.toString(QUrl.FullyEncoded)
+            if pattern.startswith('/') and pattern.endswith('/'):
+                return re.search(pattern[1:-2], display_url) != None
+            return fnmatch.fnmatch(display_url, pattern)
+
         tester = (lambda script:
-                  any(match(pat) for pat in script.includes) and
-                  not any(match(pat) for pat in script.excludes))
+                  any(_match(pat) for pat in script.includes) and
+                  not any(_match(pat) for pat in script.excludes))
```

Comment:
I'm considering moving all this matching stuff into a factory method on MatchingScripts to make scripts_for less complicated. Should I bother?

Reply:
Yeah I guess it would just be shuffling the same code around when the real problem is that the function is ugly. I've added more comments instead.